### PR TITLE
Add a notice to closed campaigns that changesets may be outdated

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes https://github.com/sourcegraph/sourcegraph/issues/16924 by
adding a notice on the CampaignsDetails page.

@eseliger I'm a bit lost when it comes to updating the required tests. Working on it.

### Screenshots

image
image

### CODE REVIEW

File: client/web/src/enterprise/campaigns/detail/ClosedNotice.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+export const ClosedNotice: React.FunctionComponent<ClosedNoticeProps> = ({ closedAt, className }) => {
+    if (!closedAt) {
+        return <></>
+    }
+    return (
+        <div className={classNames('alert alert-info', className)}>
+            <InformationOutlineIcon className="icon-inline" /> Information on this page may be out of date because
+            changesets which only exist in closed campaigns are not synced with the code host.
```
Comment (LawnGnome) [inline client/web/src/enterprise/campaigns/detail/ClosedNotice.tsx:18]: Absolute pedantry, sorry:

```suggestion
            changesets that only exist in closed campaigns are not synced with the code host.
```

**Review by LawnGnome — APPROVED** [review]
LGTM, although I'm a little surprised we don't have a storybook that picks this up.

**Review by eseliger — APPROVED** [review]
LGTM, maybe we want a story for this, too :)

File: client/web/src/enterprise/campaigns/detail/ClosedNotice.tsx (hunk 1 above)
Comment (mrnugget) [reply client/web/src/enterprise/campaigns/detail/ClosedNotice.tsx:18]: Ha! I knew it. I thought the same thing when reading the `which` in Figma, but then started doubting my non-native-English skills.

**Pull request merged**